src/metabase/public_settings/premium_features.clj

```diff
@@ -118,41 +127,68 @@
          :status        (tru "Unable to validate token")
          :error-details (tru "Token validation timed out.")}))))
 
+;;;;;;;;;;;;;;;;;;;; Airgap Tokens ;;;;;;;;;;;;;;;;;;;;
+(declare decode-airgap-token)
+
+(mu/defn ^:private max-users-allowed
+  "Returns the max users value from an airgapped key, or nil indicating there is no limt."
+  [] :- [:or pos-int? :nil]
+  (when-let [token (premium-embedding-token)]
+    (when (str/starts-with? token "airgap_")
+      (let [max-users (:max-users (decode-airgap-token token))]
+        (when (pos? max-users) max-users)))))
+
+(defn airgap-check-user-count
+  "Checks that, when in an airgap context, the allowed user count is acceptable."
+  []
+  (when-let [max-users (max-users-allowed)]
+    ;; If you add a new usery thing that is not a user, this must be updated
+    (when (>= (t2/count :model/User :is_active true, :type :personal) max-users)
```

Comment:
Should this be `>` if we're doing this check on startup? i.e. if the limit is 100 users and you're at exactly 100 that should be fine, right?

Reply:
Good catch, thanks